programs/stake/src/stake_state.rs

```diff
@@ -613,15 +613,48 @@ pub fn delegate(
     }
 }
 
+fn deactivate_stake(
+    invoke_context: &InvokeContext,
+    stake: &mut Stake,
+    stake_flags: &mut StakeFlags,
+    epoch: Epoch,
+) -> Result<(), InstructionError> {
+    let stake_history = invoke_context.get_sysvar_cache().get_stake_history()?;
+
+    if invoke_context
+        .feature_set
+        .is_active(&feature_set::stake_redelegate_instruction::id())
+        && stake_flags.contains(StakeFlags::MUST_FULLY_ACTIVATE_BEFORE_DEACTIVATION_IS_PERMITTED)
+    {
+        // when MUST_FULLY_ACTIVATE_BEFORE_DEACTIVATION_IS_PERMITTED flag is set on stake_flags,
+        // deactivation is only permitted when the stake delegation activating amount is zero.
+        let status = stake
+            .delegation
+            .stake_activating_and_deactivating(epoch, Some(&stake_history));
+        if status.activating != 0 {
+            Err(StakeError::RedelegatedStakeMustFullyActivateBeforeDeactivationIsPermitted.into())
```

Comment:
```suggestion
            Err(InstructionError::from(StakeError::RedelegatedStakeMustFullyActivateBeforeDeactivationIsPermitted))
```

`<T>::from(x)` self-documents better than `x.into()`

Reply:
Done.